### ISSUE TITLE
Timeline chart

### ISSUE THREAD

**Issue by muety** [muety/wakapi#101]
Like the one from WakaTime or Hakatime: 

![](https://raw.githubusercontent.com/mujx/hakatime/master/img/overview.png)

**Comment by muety** [comment]
Requires to introduce a concept like [durations](https://wakatime.com/developers#durations) to group heartbeats.

**Comment by muety** [comment]
Chart.js doesn't feature timeline charts as such. But using a stacked, horizontal bar chart, as suggested [here](https://stackoverflow.com/a/73071513/3112139) could be an option.

**Comment by oasido** [comment]
Is this something that's being worked on? How can I contribute?
Really missing out this feature! :P

**Comment by oasido** [comment]
Hey! What a surprise!
Great job, love the new update!!

**Comment by onns** [comment]
maybe this one? https://echarts.apache.org/examples/zh/editor.html?c=custom-profile

**Comment by Grommash9** [comment]
Will be really great to have it!

Missing it a lot from wakatime to understand when have I worked

**Comment by Grommash9** [comment]
So we don't need duration for that it can be implemented in some other way @muety ?

**Comment by muety** [comment]
Yes, we'd need (presisted / pre-aggregated) durations for this, so #716 is a prerequisites. I put that on the very top of my prio list. Stay tuned! 

Btw., thanks a lot for your dono! 🙏
